sentry/src/main/java/io/sentry/Hub.java

```diff
@@ -482,13 +479,8 @@ public void bindClient(final @NotNull ISentryClient client) {
           .log(SentryLevel.WARNING, "Instance is disabled and this 'bindClient' call is a no-op.");
     } else {
       final StackItem item = stack.peek();
-      if (client != null) {
-        options.getLogger().log(SentryLevel.DEBUG, "New client bound to scope.");
-        item.setClient(client);
-      } else {
-        options.getLogger().log(SentryLevel.DEBUG, "NoOp client bound to scope.");
-        item.setClient(NoOpSentryClient.getInstance());
-      }
```

Comment:
I guess we should keep this, `client` can be null, it's not really about the stack peek being null like the others

Reply:
`client` is marked as `@NotNull` in `StackItem`, is this incorrect?

Reply:
`client` is a method param.

that's why we set either the `client` instance or `NoOpSentryClient.getInstance()`
you're correct about `StackItem` having non-null `client`

Reply:
@marandaneto thanks, you're right. Fixed.